pkg/alertmanager/operator.go

```diff
@@ -763,10 +763,12 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 		return errors.Wrap(err, "failed to retrieve statefulset")
 	}
 
+	var current *appsv1.StatefulSet
 	oldSpec := appsv1.StatefulSetSpec{}
 	if obj != nil {
 		ss := obj.(*appsv1.StatefulSet)
 		oldSpec = ss.Spec
+		current = ss
```

Comment:
```suggestion
		current = obj.(*appsv1.StatefulSet)
		currentSpec = current.Spec
```

or even get rid of oldSpec/currentSpec?